### PR TITLE
test(sdl): improve fixture generation to address any further case

### PR DESCRIPTION
**Pull request by ygrishajev**
This PR introduces enhancements to the testability of the SDL. By leveraging the `immutability-helper` and `dot-object` libraries, we have created several functions that generate testing fixtures, such as SDL configurations, groups, and manifests.

### Key Benefits

1. **JSON as the Source Format**: plain JSON is used for source data, which offers better manageability compared to YAML. This choice simplifies the handling of data structures.
   
2. **Ease of Creating Fixtures**: The new functions make it straightforward to create new fixtures while ensuring that the source data remains unchanged. This improves the reliability and reproducibility of tests.

3. **Enhanced Tooling for JSON Mutation**: With enhanced tools to mutate JSON data, these functions provide a robust framework for adjusting test data according to specific test requirements.

4. **Future-proofing**: Designed to accommodate nearly any scenario that might arise in future developments, this approach enhances the scalability and adaptability of our testing infrastructure.

### CODE REVIEW

File: src/sdl/SDL/SDL.spec.ts [hunk 1]
```diff
@@ -201,24 +193,34 @@ describe("SDL", () => {
...
         expect(() => {
-          SDL.fromString(readBasicSdl({ credentials: omit(credentials, field) }), "beta3", "sandbox");
+          SDL.fromString(yml, "beta3", "sandbox");
         }).toThrowError(new SdlValidationError(`service "web" credentials missing "${field}"`));
       });
 
       it.each(fields)('should throw an error when credentials "%s" is empty', field => {
         credentials[field] = "";
+        const yml = createSdlYml({
+          "services.web.credentials": { $set: credentials }
```
Comment (baktun14) [inline src/sdl/SDL/SDL.spec.ts:208]: Pretty cool, I never used `immutability-helper` per se. Is it strongly typed?

**Review by baktun14 — APPROVED** [review]

File: src/sdl/SDL/SDL.spec.ts (hunk 1 above)
Comment (ygrishajev) [reply src/sdl/SDL/SDL.spec.ts:208]: Yes. However dot-object is not. I still used it as it's for tests and it simplified code quite a bit. If we feel like it we could add own typing there.

Comment (ygrishajev) [reply src/sdl/SDL/SDL.spec.ts:208]: Immutability helper is generally used to ui state management :) but it's useful in this case it seems

Comment (baktun14) [reply src/sdl/SDL/SDL.spec.ts:208]: Yea, I used immutablejs a lot in the past

**Pull request merged**